File: agroguardian_app_ios/agroguardian_app_ios_watch Watch App/Models/CommandData.swift
import Foundation
import SwiftUI

struct CommandData: Decodable, Identifiable {
    let id: Int
    let type: String
    let timestamp: String

    private static let isoFormatter: ISO8601DateFormatter = {
        let formatter = ISO8601DateFormatter()
        return formatter
    }()

    var time: Date {
        CommandData.isoFormatter.date(from: timestamp) ?? Date()
    }
} 

File: agroguardian_app_android/app/src/main/java/com/example/agroguardianappandroid/data/models/CommandData.kt
package com.example.agroguardianappandroid.data.models

import java.text.SimpleDateFormat
import java.util.Date
import java.util.Locale

data class CommandData(
    val id: Int,
    val type: String,
    val timestamp: String
) {
    private val isoFormatter = SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss'Z'", Locale.getDefault())

    val time: Date
        get() {
            return try {
                isoFormatter.parse(timestamp) ?: Date()
            } catch (e: Exception) {
                Date()
            }
        }
}